test/Quoter.t.sol

```diff
@@ -121,9 +121,9 @@ contract QuoterTest is Test, Deployers {
 
     // nested self-call into lockAcquired reverts
     function testQuoter_callLockAcquired_reverts() public {
-        vm.expectRevert(IQuoter.InvalidLockAcquiredSender.selector);
+        vm.expectRevert(IQuoter.LockFailure.selector);
```

Comment:
Is this the expected behaviour? I don't see why we are actually getting this error vs the previous after removing the callerAddress from the function signature.

Reply:
this is a weird test. its testing re-entrancy of lockAcquired?

```solidity
// initiate a reentrancy call
quoter.lockAcquired(abi.encodeWithSelector(quoter.lockAcquired.selector, address(this), "0x"));

// inside lockAcquired, we are re-entering into a 2nd lockAcquired call
address(this).call(...);

// inside the 2nd lockAcquired call, we revert since the sender is the Quoter itself (1st lockAcquired)
if (msg.sender != address(manager)) {
    revert InvalidLockAcquiredSender();
}

// returning to the first lockAcquired call:
(bool success, bytes memory returnData) = address(this).call(data);
if (success) return returnData;
if (returnData.length == 0) revert LockFailure();
```

its the last LockFailure that is getting caught by the test

Reply:
Hm maybe we should be bubbling up the InvalidLockAcquiredSender() revert then?